web-ui/components/Games/Maze.vue

```diff
@@ -13,159 +13,47 @@
-            "Open",
-            "Open",
-            "Open",
-            "Open",
-            "Open",
-          ],
-          [
-            "Open",
-            "Open",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-          ],
-          [
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Blocked",
-            "Open",
-          ],
-        ],
-      };
-    },
-    mapMaze() {
-      const map = this.gameState.map;
-      const [playerX, playerY] = [
-        this.gameState.player.x,
-        this.gameState.player.y,
-      ];
-      const [exitX, exitY] = [this.gameState.exit.x, this.gameState.exit.y];
+      },
+      gameState: {
+        visualMap: this.emptyMaze(),
+      },
+      emptyMap: this.emptyMaze(),
+    };
+  },
+  updated() {
+    this.getMaze();
+  },
+  methods: {
+    async getMaze() {
```

Comment:
I don't think that any of the fetching the maze logic should exist in this component. It makes it harder to update from the outside based on different triggers. It would be cleaner to have just the code associated with drawing the maze located here and the maze as a whole passed in as a prop.

This way the maze can be updated externally by simply replacing the prop data. With your method the starting of a new game is tightly coupled to this component, instead it should be the responsibility of the run method: https://github.com/mdaffin/sprockit.io/blob/master/web-ui/pages/index.vue#L59. Then that method can simple tell this component what to render by passing down the maze when the game has completed.